build-tools/src/test/groovy/org/elasticsearch/gradle/DistributionDownloadPluginSpec.groovy

```diff
@@ -0,0 +1,121 @@
+import static org.elasticsearch.gradle.distribution.ElasticsearchDistributionTypes.ARCHIVE
+import static org.elasticsearch.gradle.distribution.ElasticsearchDistributionTypes.INTEG_TEST_ZIP
+
+class DistributionDownloadPluginSpec extends Specification {
+
+    private Project project
+
+    def setup() {
+        Project rootProject = ProjectBuilder.builder().build();
+        project = ProjectBuilder.builder().withParent(rootProject).build()
+        project.plugins.apply("elasticsearch.distribution-download")
+    }
+
+    def "version default provides"() {
+        when:
+        def distro = distro(project, "testdistro", null, ARCHIVE, LINUX, true)
+        then:
+        distro.version == VersionProperties.elasticsearch
+    }
+
+    def "bad version format is handled"() {
+        when:
+        distro(project, "testdistro", "badversion", ARCHIVE, LINUX, true)
+        then:
+        def e = thrown(IllegalArgumentException)
+        e.message == "Invalid version format: 'badversion'. Should be major.minor.revision[-(alpha|beta|rc)Number|-SNAPSHOT]"
+    }
+
+    def "type default provided"() {
+        when:
+        def distro = distro(project, "testdistro", "5.0.0", null, LINUX, true)
+        then:
+        distro.type == ARCHIVE
+    }
+
+    def "platform default provided"() {
+        when:
+        def distro = distro(project, "testdistro", "5.0.0", ARCHIVE, null, true)
+        then:
+        distro.platform == CURRENT_PLATFORM
+    }
+
+    def "platform for integTest is not configurable"() {
+        when:
+        distro(project, "testdistro", "5.0.0", INTEG_TEST_ZIP, LINUX, null)
+        then:
+        def e = thrown(IllegalArgumentException)
+        e.message == "platform cannot be set on elasticsearch distribution [testdistro] of type [integ_test_zip]"
+    }
+
+    def "bundled jdk default provided"() {
+        when:
+        def distro = distro(project, "testdistro", "5.0.0", ARCHIVE, LINUX, true)
+        then:
+        distro.bundledJdk
+    }
+
+    def "bundled jdk for integTest is not configurable"() {
+        when:
+        distro(project,  "testdistro", "5.0.0", INTEG_TEST_ZIP, null, true)
+        then:
+        def e = thrown(IllegalArgumentException)
+        e.message == "bundledJdk cannot be set on elasticsearch distribution [testdistro] of type [integ_test_zip]"
+    }
+
+    // create a distro and finalize its configuration
+    private ElasticsearchDistribution distro(
+            Project project,
+            String name,
+            String version,
+            ElasticsearchDistributionType type,
+            ElasticsearchDistribution.Platform platform,
+            Boolean bundledJdk
+    ) {
+        ElasticsearchDistribution distribution = createDistro(project, name, version, type, platform, bundledJdk);
+        distribution.finalizeValues();
+        return distribution;
+    }
+
+    private ElasticsearchDistribution createDistro(
```

Comment:
Any reason not to inline this into `distro(...)` above?

Reply:
done